### PR TITLE
[WIP] IBM MQ source

### PR DESCRIPTION
**Pull request by sebgoa**
This is a proposed new container source, it gets events from IBM MQ queues.

It is self-contained and moved as is in `/contrib`

It does not use `ko` yet.

### CODE REVIEW

File: contrib/ibm-mq/Makefile [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+	rm -rf $(TARGET)
+
+fmt:
+	go fmt ./...
+
+lint:
+	golint ./...
+
+vet:
+	go vet ./...
```
Comment (mattmoor-sockpuppet) [inline contrib/ibm-mq/Makefile]:
```suggestion
	go vet ./...

```

File: contrib/ibm-mq/README.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+If you want to add and remove message in bulk, build the Docker image for IBM MQ Local client from `local-client` folder and follow the instructions from the tutorial (step 5) to get messages to the MQ (enter `put 20` in the shell prompt)
+
+### Knative usage
+
+Edit the Container source manifest and apply it:
+
+```
+kubectl apply -f mq-source.yaml
+```
```
Comment (mattmoor-sockpuppet) [inline contrib/ibm-mq/README.md]:
```suggestion
```

```

File: contrib/ibm-mq/mq-source.yaml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+      value: QM1
+    - name: CHANNEL_NAME
+      value: DEV.ADMIN.SVRCONN
+    - name: CONNECTION_NAME
+      value: mq-service.default.svc.cluster.local(1414)
+    - name: USER_ID
+      value: admin
+    - name: PASSWORD
+      value: passw0rd
+    - name: QUEUE_NAME
```
Comment (mattmoor-sockpuppet) [inline contrib/ibm-mq/mq-source.yaml]:
```suggestion
    - name: QUEUE_NAME
```

**Comment by evankanderson** [comment]
I suggest you add an OWNERS here pointing to yourself, so that you and co-maintainers can review subsequent updates without needing upstream reviewers.

See https://github.com/knative/eventing-sources/blob/master/contrib/kafka/OWNERS and https://github.com/knative/eventing-sources/blob/master/OWNERS_ALIASES for an example

File: contrib/ibm-mq/Makefile (hunk 1 above)
Comment (mattmoor-sockpuppet) [inline contrib/ibm-mq/Makefile]:
```suggestion
	go vet ./...

```

File: contrib/ibm-mq/README.md (hunk 2 above)
Comment (mattmoor-sockpuppet) [inline contrib/ibm-mq/README.md]:
```suggestion
```

```

File: contrib/ibm-mq/mq-source.yaml (hunk 3 above)
Comment (mattmoor-sockpuppet) [inline contrib/ibm-mq/mq-source.yaml]:
```suggestion
    - name: QUEUE_NAME
```

File: contrib/ibm-mq/README.md (hunk 2 above)
Comment (mattmoor-sockpuppet) [inline contrib/ibm-mq/README.md]:
```suggestion
```

```

File: contrib/ibm-mq/mq-source.yaml (hunk 3 above)
Comment (mattmoor-sockpuppet) [inline contrib/ibm-mq/mq-source.yaml]:
```suggestion
    - name: QUEUE_NAME
```

File: contrib/ibm-mq/README.md (hunk 2 above)
Comment (mattmoor-sockpuppet) [inline contrib/ibm-mq/README.md]:
```suggestion
```

```

**Comment by sebgoa** [comment]
@evankanderson thanks for looking. I added a OWNERS file and modified the OWNER_ALIASES file.
Fixed the copyright.

Because this needs some bindings, I don't see how we can use `ko` for this, so we have to add a Dockerfile.

I will need to figure out how we can tie this to automatically pushing a container. Right now we are referencing a container in `gcr.io/triggermesh` project.

I know @duglin is having a look at this as well.

**Comment by vaikas-google** [comment]
Looks like it's maybe missing some dependencies?

I0604 20:17:22.470] Running tests with 'go test -race -v  ./... '
I0604 20:17:24.992] contrib/ibm-mq/main.go:25:2: cannot find package "github.com/caarlos0/env" in any of:
I0604 20:17:24.995] 	/go/src/github.com/knative/eventing-sources/vendor/github.com/caarlos0/env (vendor tree)
I0604 20:17:24.995] 	/usr/local/go/src/github.com/caarlos0/env (from $GOROOT)
I0604 20:17:24.995] 	/go/src/github.com/caarlos0/env (from $GOPATH)
I0604 20:17:24.998] contrib/ibm-mq/main.go:26:2: cannot find package "github.com/ibm-messaging/mq-golang/ibmmq" in any of:
I0604 20:17:25.001] 	/go/src/github.com/knative/eventing-sources/vendor/github.com/ibm-messaging/mq-golang/ibmmq (vendor tree)
I0604 20:17:25.002] 	/usr/local/go/src/github.com/ibm-messaging/mq-golang/ibmmq (from $GOROOT)
I0604 20:17:25.002] 	/go/src/github.com/ibm-messaging/mq-golang/ibmmq (from $GOPATH)
I0604 20:17:25.003] contrib/ibm-mq/main.go:27:2: cannot find package "github.com/knative/pkg/cloudevents" in any of:
I0604 20:17:25.003] 	/go/src/github.com/knative/eventing-sources/vendor/github.com/knative/pkg/cloudevents (vendor tree)
I0604 20:17:25.004] 	/usr/local/go/src/github.com/knative/pkg/cloudevents (from $GOROOT)
I0604 20:17:25.005] 	/go/src/github.com/knative/pkg/cloudevents (from $GOPATH)
I0604 20:17:25.021] Finished run, return code is 1
I0604 20:17:25.063] XML report written to /workspace/_artifacts/junit_IIOBcBiN.xml

**Comment by sebgoa** [comment]
This does not build with `ko`, we need to run `dep ensure` to get the dependencies before the `go build`

File: contrib/ibm-mq/README.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+docker run --rm -e PASSWORD=password \
+                --net=container:mq \
+                mqsource --sink http://localhost:8080
+```
+
+1. Open the MQ console:
+
+Using _Admin_ and _passw0rd_ as default development credentials do:
+
+`open https://localhost:9443/ibmmq/console/`
```
Comment (mattmoor-sockpuppet) [inline contrib/ibm-mq/README.md]:
```suggestion
`open https://localhost:9443/ibmmq/console/`
```

File: contrib/ibm-mq/README.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+  }
+```
+
+### Knative usage
+
+Edit the Container source manifest to specify your queue and connection parameters then and apply it like so:
+
+```
+kubectl apply -f mq-source.yaml
+```
```
Comment (mattmoor-sockpuppet) [inline contrib/ibm-mq/README.md]:
```suggestion
```

```

File: contrib/ibm-mq/README.md [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+docker run --rm -e PASSWORD=password \
+                --net=container:mq \
+                mqsource --sink http://localhost:8080
+```
+
+1. Open the MQ console:
+
+Using _Admin_ and _passw0rd_ as default development credentials do:
+
+`open https://localhost:9443/ibmmq/console/`
```
Comment (mattmoor-sockpuppet) [inline contrib/ibm-mq/README.md]:
```suggestion
`open https://localhost:9443/ibmmq/console/`
```

File: contrib/ibm-mq/README.md [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+  }
+```
+
+### Knative usage
+
+Edit the Container source manifest to specify your queue and connection parameters then and apply it like so:
+
+```
+kubectl apply -f mq-source.yaml
+```
```
Comment (mattmoor-sockpuppet) [inline contrib/ibm-mq/README.md]:
```suggestion
```

```

**Comment by sebgoa** [comment]
closing in favor of #500

**Pull request closed without merging**